x/paramfilter/test/gov_params_test.go

```diff
@@ -0,0 +1,604 @@
+				want := uint32(1)
+				assert.Equal(want, got)
+			},
+		},
+		{
+			"staking.MaxEntries",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyMaxEntries),
+				Value:    `1`,
+			}),
+			func() {
+				got := suite.app.StakingKeeper.GetParams(suite.ctx).MaxEntries
+				want := uint32(1)
+				assert.Equal(want, got)
+			},
+		},
+		{
+			"staking.MaxValidators",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyMaxValidators),
+				Value:    `1`,
+			}),
+			func() {
+				got := suite.app.StakingKeeper.GetParams(suite.ctx).MaxValidators
+				want := uint32(1)
+				assert.Equal(want, got)
+			},
+		},
+		{
+			"staking.MinCommissionRate",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyMinCommissionRate),
+				Value:    `"1"`,
+			}),
+			func() {
+				got := suite.app.StakingKeeper.GetParams(suite.ctx).MinCommissionRate
+				want := sdk.NewDec(1)
+				assert.Equal(want, got)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			err := suite.govHandler(suite.ctx, tc.proposal)
+			suite.Require().NoError(err)
+			tc.postProposal()
+		})
+	}
+}
+
+// TestUnmodifiableParams verifies that the params listed as non governance
+// modifiable in the specs params.md file cannot be modified via
+// governance.
+func (suite *GovParamsTestSuite) TestUnmodifiableParams() {
+	assert := suite.Assert()
+
+	testCases := []struct {
+		name         string
+		proposal     *proposal.ParameterChangeProposal
+		expectErr    bool
+		postProposal func()
+	}{
+		{
+			"bank.SendEnabled",
+			testProposal(proposal.ParamChange{
+				Subspace: banktypes.ModuleName,
+				Key:      string(banktypes.KeySendEnabled),
+				Value:    `[{"denom": "test", "enabled": false}]`,
+			}),
+			true,
+			func() {
+				got := suite.app.BankKeeper.GetParams(suite.ctx).SendEnabled
+				proposed := []*banktypes.SendEnabled{banktypes.NewSendEnabled("test", false)}
+				assert.NotEqual(proposed, got)
+			},
+		},
+		{
+			"consensus.block.TimeIotaMs",
+			testProposal(proposal.ParamChange{
+				Subspace: baseapp.Paramspace,
+				Key:      string(baseapp.ParamStoreKeyBlockParams),
+				Value:    `{"max_bytes": "1", "max_gas": "1", "time_iota_ms": "1"}`,
+			}),
+			// TimeIotaMs is not blocked by the param filter but it isn't
+			// exposed from CometBFT to the application so it can not be
+			// modified by governance. Since it isn't blocked by the param
+			// filter, we expect no error if a governance proposal includes it.
+			// We do expect the value to not change.
+			false,
+			func() {
+				got := suite.app.BaseApp.GetConsensusParams(suite.ctx).Block
+				proposed := tmproto.BlockParams{
+					MaxBytes:   1,
+					MaxGas:     1,
+					TimeIotaMs: 1,
+				}
+				assert.NotEqual(proposed, got)
```

Comment:
It would be advantageous to enhance the tests by verifying that the parameter values before and after submitting a proposal remain consistent for non-modifiable parameters. This is in addition to ensuring that the updated values do not match the proposed changes.

Reply:
I agree completely. I thought of that while working on the test but didn't end up implementing. Can do though

Reply:
Resolved via dd0585ccce018359982a98bf9bfa379e289b24c0